apps/nextjs-website/src/app/[productSlug]/tutorials/[...productTutorialPage]/page.tsx

```diff
@@ -26,6 +27,34 @@ type ProductTutorialPageProps = {
   bodyConfig: ParseContentConfig;
 } & LayoutProps;
 
+export async function generateMetadata({
+  params,
+}: {
+  params: Params;
+}): Promise<Metadata> {
+  const productSlug = params?.productSlug;
+  const tutorialPath = params?.productTutorialPage?.join('/');
+  const tutorialProps = await getTutorial(productSlug, [tutorialPath]);
+  const { page } = tutorialProps;
+
+  const body = page.body;
+  const lines = body.split('\n').filter((line) => line !== '');
+  const title = lines[0].replace('# ', '');
```

Comment:
Same as previous comment: why these lines are needed?
Moreover, it looks like there are the very same line somewhere else, so wouldn't be better to extract those lines in a function and test the function as well?